Arranged "backup" sections into one topic

bfallonf opened the pull request:
First attempt at putting the backup docs into one place. Note that I didn't take the sections out of the old locations, but copied them (modularization!).

@kalexand-rh WDYT? Ignore the pathetic intro and pay more attention to the the structure, please, and tell me if you were imagining something different.

bfallonf commented:
It ain't pretty for some reason, but here's a preview: http://file.bne.redhat.com/bfallonf/08_06_2018/day_two_guide/environment_backup.html

kalexand-rh commented:
@bfallonf structurally, yes, that looks like what I'd expect for a new back up topic. :)
Before you merge, I think you should edit the modules to show that they are used in the new assembly.
Let me know if you want a peer review on the new text or comments about that intro.

bfallonf commented:
@kalexand-rh Oof this took a bit. I've updated the assembly to contain a better intro, and the individual modules to say which assembly they're going into. Feel free to review.

How do you want to proceed? I'm guessing you're in the process of finding info for a "restore" topic to add to the day two guide? I'd like a link to that from the intro I've added once it's done (no rush).

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+$ journalctl -fu etcd.service
+----
+
+. Repeat the above for every etcd node joining the cluster.
```
s/the above/the previous steps

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+
+. Repeat the above for every etcd node joining the cluster.
+
+. Verify the cluster status and cluster health once all the nodes joined:
```
. After you add all the nodes to the cluster, verify the cluster status and cluster health:

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+
+==== Restoring etcd for v3
+
+The restore procedure for v3 data is similar to the v2 data.
```
s/similar to the v2 data./similar to the restore procedure for the v2 data.

Are there restrictions or guidelines for when you restore one, the other, or both types of data?

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+
+The restore procedure for v3 data is similar to the v2 data.
+
+Snapshot integrity may be optionally verified at restore time. If the snapshot
```
I think this information should go with the restore command.

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+
+[IMPORTANT]
+====
+The procedure to restore only the v3 data must be performed on a single etcd
```
What about the v2 data?

Maybe "You can restore the v3 data on only a single etcd host. You can add more etcd nodes after you restore the first node." And add a link to the section about adding etcd hosts.

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+# rm -Rf /var/lib/etcd
+----
+
+. Use the `snapshot restore` command with the data from `/etc/etcd/etcd.conf` to
```
S/Use the `snapshot restore` command with the data from `/etc/etcd/etcd.conf` to match the following command:/ Run the `snapshot restore` command. Substitute values from the `/etc/etcd/etcd.conf` file in your etcd snapshot.

Is the peer url right?

kalexand-rh commented on day_two_guide/environment_backup.adoc:
```diff
@@ -0,0 +1,43 @@
+toc::[]
+
+Creating an environment-wide backup involves copying important data
+in case of crashing instances, or corrupt data. Once backups have been created,
```
s/Once/After (Style says: once adv Use only to mean “one time.” Do not use as a conjunction to mean “after” or “when.”)

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+
+*Adding more nodes*
+
+Once the first instance is running, it is safe to restore multiple etcd servers as desired.
```
s/Once/After
s/it is safe to restore multiple etcd servers as desired./you can add more etcd nodes to your cluster.

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+# grep ETCD_NAME /etc/etcd/etcd.conf
+----
+
+. Get the IP where etcd listens for peer communication:
```
s/IP/IP address

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+  --peer-urls="*<advertise_peer_urls>*"
+----
+
+. The command outputs some variables. For example:
```
Same as 189.

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+# vi /etc/etc/etcd.conf
+----
+
+. In the recently added etcd node, clean the etcd data directories to ensure the
```
Same as 41.

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+# journalctl -fu etcd.service
+----
+
+. Repeat the previous steps for every etcd node that is required to be added.
```
s/that is required to be added./you want to add.

kalexand-rh commented on day_two_guide/topics/restore_etcd.adoc:
```diff
@@ -0,0 +1,382 @@
+
+. Repeat the previous steps for every etcd node that is required to be added.
+
+. Verify the cluster has been properly set:
```
I'd make this like 112.

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+////
+
+Scaling the etcd cluster can be performed vertically by adding more resources to
+the etcd hosts, or horizontally by adding more etcd hosts.
```
s/Scaling the etcd cluster can be performed/You can scale the etcd cluster

no comma

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+number of members.
+====
+
+The new host requires a fresh RHEL7 dedicated host. The etcd storage should be
```
Should these be prereqs?
I'd spell out RHEL
I'd replace the second sentence with something like "For best performance, use a dedicated SSD disk that is mounted in `/var/lib/etcd` for the etcd storage."

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+[discrete]
+== Procedure
+
+. Modify the Ansible inventory file and create a new group named `[new_etcd]`
```
maybe s/file and create/file to create

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+$ ansible-playbook  /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-etcd/scaleup.yml
+----
+
+. Once the above has finished, modify the inventory file to reflect the current
```
s/Once the above has finished/After the playbook runs,

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+etcd0.example.com
+----
+
+. If using Flannel, modify the `flanneld` service configuration, located at
```
I'd make this like line 125 of day_two_guide/topics/removing_etcd_host.adoc

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+The number of etcd nodes must be odd, so at least two hosts must be added.
+====
+
+. If using Flannel, the `flanneld` service configuration located at
```
same as 135

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+
+==== Manually adding a new etcd host
+
+The following steps can be performed on any etcd member. If using the Ansible
```
So, do we perform these steps on any etcd member or the first host in the Ansible inventory?
Which Ansible installer are you talking about?

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+== Procedure
+
+. In order to create the etcd certificates, run the `openssl` command with the
+proper values. To make this process easier, create some environment variables:
```
What are the "proper values?"

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+for more information.
+====
+
+. Create the directory where the configuration and certificates are stored:
```
. Create a directory to store the configuration files and certificates:

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+----
+
+. Copy the current etcd configuration and `ca.crt` files from the current node
+as examples to be modified later:
```
s/to be modified/to modify

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# cp /etc/etcd/ca.crt ${PREFIX}
+----
+
+. Add the new host to the etcd cluster. Note the new host is not configured yet
```
Would the bit about the status make more sense somewhere else? I don't see where that's displayed in the output.

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+as examples to be modified later:
++
+----
+# cp /etc/etcd/etcd.conf ${PREFIX}
```
What do you mean by ${PREFIX}?

bfallonf replied:
It's one of the variables set at the start

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+----
+
+. Those values must be overwritten by the current ones in the sample
+`${PREFIX}/etcd.conf` file. Also, modify the following variables with the new
```
Maybe 
. Modify the sample `etcd.conf` file that you copied. Overwrite its parameter values with the values in the output of the last command and provide the new host IP address by using the `${NEW_ETCD_IP}` environment variable.

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+ETCD_ADVERTISE_CLIENT_URLS
+----
+
+. Modify the `${PREFIX}/etcd.conf` file and check for syntax errors or missing
```
s/could/might
I'd say "Review the `${PREFIX}/etcd.conf` file for syntax errors or missing IP addresses:"

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# vi ${PREFIX}/etcd.conf
+----
+
+. Once the file has been properly modified, a `tgz` file with the certificates,
```
./ Create a `tgz` file that contains the certificates and configuration files and copy it to the new etcd host:

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+
+*Steps to be performed on the new etcd host*
+
+The new host is required to be subscribed to the proper Red Hat software
```
Do we need to say this again?

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# yum install -y iptables-services
+----
+
+. Create firewall rules to allow etcd to communicate:
```
maybe

. Create the `OS_FIREWALL_ALLOW` filewall rules to allow etcd to communicate with:

* Clients on port 2379/tcp
* etcd peers on port 2380/tcp

the note about the chain

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# rm -Rf /var/lib/etcd/*
+----
+
+. Untar the certificates and configuration files
```
s/Untar/Extract
s/files/files:

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+for the instance to allow incoming traffic to those ports as well.
+====
+
+. Install etcd software:
```
s/etcd software:/etcd:

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# yum install -y etcd
+----
+
+. Ensure the service is not running:
```
s/service/etcd service

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# tar xzvf /tmp/etcd0.example.com.tgz -C /etc/etcd/
+----
+
+. Restore etcd configuration and data owner:
```
What?

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+# systemctl enable etcd --now
+----
+
+. Verify the host has been added to the cluster and the current cluster health:
```
Like day_two_guide/topics/restore_etcd.adoc line 112

kalexand-rh commented on day_two_guide/topics/scaling_etcd.adoc:
```diff
@@ -0,0 +1,432 @@
+    - https://etcd0.example.com:2379
+----
+
+. Restart the master API service on every master:
```
I'd make these choices.

kalexand-rh commented on day_two_guide/host_level_tasks.adoc:
```diff
@@ -69,7 +69,22 @@ include::day_two_guide/topics/node_maintenance.adoc[]
 [[day-two-guide-etcd-backup]]
 == etcd tasks
 
+=== etcd backup
```
Will you add this text to line 11 for me?
```
:context: day_two_host_level_tasks
```

kalexand-rh commented on day_two_guide/topics/removing_etcd_host.adoc:
```diff
@@ -0,0 +1,163 @@
+# systemctl restart flanneld.service
+----
+
+==== Replacing an etcd host
```
I think this should go at the top of this module.

bfallonf commented on day_two_guide/environment_backup.adoc:
```diff
@@ -0,0 +1,43 @@
+in case of crashing instances, or corrupt data. Once backups have been created,
+they can be restored onto a newly installed version of the relevant component.
+
+Backups should be performed on a regular basis to prevent data loss.
```
Hmm you're right. Changed to: Perform a back up on a regular basis to prevent data loss.

bfallonf commented on day_two_guide/topics/removing_etcd_host.adoc:
```diff
@@ -0,0 +1,163 @@
++
+[NOTE]
+====
+The `remove` command requires the etcd ID, not the hostname.
```
We're saying _not_ to use it, so that might give the wrong impression.

bfallonf commented:
Thanks @kalexand-rh . I took in some of the comments, but skipped over some of the nits. Too many to comment on individually...

I'll merge.

bfallonf commented:
/cherrypick enterprise-3.10

bfallonf commented:
/cherrypick enterprise-3.9

bfallonf commented:
/cherrypick enterprise-3.7
